lib/source-code/source-code.js

```diff
@@ -921,6 +971,84 @@ class SourceCode extends TokenStore {
         return configNodes;
     }
 
+    /**
+     * Returns an all directive nodes that enable or disable rules along with any problems
+     * encountered while parsing the directives.
+     * @returns {{problems:Array<Problem>,directives:Array<Directive>}} Information
+     *      that ESLint needs to further process the directives.
+     */
+    getDisableDirectives() {
+
+        // check the cache first
+        const cachedDirectives = this[caches].get("disableDirectives");
+
+        if (cachedDirectives) {
+            return cachedDirectives;
+        }
+
+        const problems = [];
+        const directives = [];
+
+        this.getInlineConfigNodes().forEach(comment => {
+            const { directivePart, justificationPart } = commentParser.extractDirectiveComment(comment.value);
+
+            // Step 1: Extract the directive text
+            const match = directivesPattern.exec(directivePart);
+
+            if (!match) {
+                return;
+            }
+
+            const directiveText = match[1];
+
+            // Step 2: Extract the directive value
+            const lineCommentSupported = /^eslint-disable-(next-)?line$/u.test(directiveText);
+
+            if (comment.type === "Line" && !lineCommentSupported) {
+                return;
+            }
+
+            // Step 3: Validate the directive does not span multiple lines
+            if (directiveText === "eslint-disable-line" && comment.loc.start.line !== comment.loc.end.line) {
+                const message = `${directiveText} comment should not span multiple lines.`;
+
+                problems.push({
+                    ruleId: null,
+                    message,
+                    loc: comment.loc
+                });
+                return;
+            }
+
+            // Step 4: Extract the directive value and create the Directive object
+            const directiveValue = directivePart.slice(match.index + directiveText.length);
+
+            switch (directiveText) {
+                case "eslint-disable":
+                case "eslint-enable":
+                case "eslint-disable-next-line":
+                case "eslint-disable-line": {
+                    const directiveType = directiveText.slice("eslint-".length);
+
+                    directives.push(new Directive({
+                        type: directiveType,
+                        node: comment,
+                        value: directiveValue,
+                        justification: justificationPart
+                    }));
```

Comment:
To confirm: the previous implementation had `problems.push(... .directiveProblems);` here; its was removed now, was that intentional?

Reply:
Yes. Those problems were created via `createDisableDirectives()` in linter. When I moved the functionality here, that function is no longer accessible. It is run inside of linter: https://github.com/eslint/eslint/pull/18328/files#diff-a4ade4bc7a7214733687d84fbb54b625e464d13be7181caf54f564e5985db980R559